src/view3d.cpp

```diff
@@ -185,31 +212,132 @@ void View3D::centerOnGeometry(const QModelIndex& index)
     m_camera.setEyeToCenterDistance(std::max<double>(2*m_camera.clipNear(), diag*0.7));
 }
 
+// #define GL_CHECK_RENDER
+#ifdef GL_CHECK_RENDER
+
+void _glError(const char *file, int line) {
+    GLenum err (glGetError());
+
+    while(err!=GL_NO_ERROR) {
+        std::string error;
+
+        switch(err) {
+            case GL_INVALID_OPERATION:      error="INVALID_OPERATION";      break;
+            case GL_INVALID_ENUM:           error="INVALID_ENUM";           break;
+            case GL_INVALID_VALUE:          error="INVALID_VALUE";          break;
+            case GL_OUT_OF_MEMORY:          error="OUT_OF_MEMORY";          break;
+            case GL_INVALID_FRAMEBUFFER_OPERATION:  error="INVALID_FRAMEBUFFER_OPERATION";  break;
+        }
+
+        tfm::printfln("GL_%s - %s:%i", error, file, line);
+        err=glGetError();
+    }
+}
+
+#define glCheckError() _glError(__FILE__,__LINE__)
+
+void _glFrameBufferStatus(GLenum target, const char *file, int line) {
+
+    GLenum fb_status(glCheckFramebufferStatus(target));
+
+    std::string status;
+
+    switch(fb_status) {
+        case GL_FRAMEBUFFER_COMPLETE:   status="COMPLETE";      break;
+        case GL_FRAMEBUFFER_UNDEFINED:  status="UNDEFINED";     break;
+        case GL_FRAMEBUFFER_INCOMPLETE_ATTACHMENT: status="INCOMPLETE_ATTACHMENT";      break;
+        //case GL_FRAMEBUFFER_INCOMPLETE_MISSING_ATTACHMENT: status="INCOMPLETE_MISSING_ATTACHMENT";      break;
+        case GL_FRAMEBUFFER_INCOMPLETE_DRAW_BUFFER: status="INCOMPLETE_DRAW_BUFFER";      break;
+        case GL_FRAMEBUFFER_INCOMPLETE_READ_BUFFER: status="INCOMPLETE_READ_BUFFER";      break;
+        case GL_FRAMEBUFFER_INCOMPLETE_MULTISAMPLE: status="INCOMPLETE_MULTISAMPLE";      break;
+        case GL_FRAMEBUFFER_INCOMPLETE_LAYER_TARGETS: status="INCOMPLETE_LAYER_TARGETS";  break;
+        //case GL_FRAMEBUFFER_INCOMPLETE_DIMENSIONS: status="INCOMPLETE_DIMENSIONS";      break;
+        case GL_FRAMEBUFFER_INCOMPLETE_MISSING_ATTACHMENT: status="INCOMPLETE_MISSING_ATTACHMENT";      break;
+        case GL_FRAMEBUFFER_UNSUPPORTED: status="UNSUPPORTED";      break;
+        default:                        status="INCOMPLETE";    break;
+    }
+
+    tfm::printfln("GL_FRAMEBUFFER_%s - %s:%i", status, file, line);
+}
+
+#define glFrameBufferStatus(TARGET) _glFrameBufferStatus(TARGET, __FILE__,__LINE__)
+
+#endif //GL_CHECK_RENDER
```

Comment:
I think it's also worth moving this stuff into glutil.h since it seems like generally useful utility code.